tests/converters/app-converter.test.ts

```diff
@@ -0,0 +1,308 @@
+import { AppConverter, StackConverter } from '../../src/converters/app-converter';
+import { Stack } from 'aws-cdk-lib/core';
+import { StackComponentResource, StackOptions } from '../../src/types';
+import * as path from 'path';
+import * as mockfs from 'mock-fs';
+import * as pulumi from '@pulumi/pulumi';
+import { MockCallArgs, MockResourceArgs } from '@pulumi/pulumi/runtime';
+import { Bucket, BucketPolicy } from '@pulumi/aws-native/s3';
+import { createStackManifest } from '../utils';
+
+// Convert a pulumi.Output to a promise of the same type.
+function promiseOf<T>(output: pulumi.Output<T>): Promise<T> {
+    return new Promise((resolve) => output.apply(resolve));
+}
+
+function setMocks() {
+    pulumi.runtime.setMocks(
+        {
+            call: (args: MockCallArgs): { [id: string]: any } => {
+                switch (args.token) {
+                    case 'aws-native:index:getAccountId':
+                        return {
+                            accountId: '12345678910',
+                        };
+                    case 'aws-native:index:getRegion':
+                        return {
+                            region: 'us-east-2',
+                        };
+                    case 'aws-native:index:getPartition':
+                        return {
+                            partition: 'aws',
+                        };
+                    case 'aws-native:index:getAzs':
+                        return {
+                            azs: ['us-east-1a', 'us-east-1b'],
+                        };
+                    default:
+                        return {};
+                }
+            },
+            newResource: (args: MockResourceArgs): { id: string; state: any } => {
+                return {
+                    id: args.name + '_id',
+                    state: {
+                        ...args.inputs,
+                        arn: args.name + '_arn',
+                    },
+                };
+            },
+        },
+        'project',
+        'stack',
+        false,
+    );
+}
+
+class MockStackComponent extends StackComponentResource {
+    public readonly name = 'stack';
+    public readonly assemblyDir: string;
+    public stack: Stack;
+    public options?: StackOptions | undefined;
+    constructor(dir: string) {
+        super('stack');
+        this.assemblyDir = dir;
+        this.registerOutputs();
+    }
+
+    registerOutput(outputId: string, output: any): void {}
+}
+
+describe('App Converter', () => {
+    const manifestFile = '/tmp/foo/bar/does/not/exist/manifest.json';
+    const manifestStack = '/tmp/foo/bar/does/not/exist/test-stack.template.json';
+    const manifestTree = '/tmp/foo/bar/does/not/exist/tree.json';
+    const manifestAssets = '/tmp/foo/bar/does/not/exist/test-stack.assets.json';
+    beforeEach(() => {
+        mockfs({
+            // Recursively loads all node_modules
+            node_modules: mockfs.load(path.resolve(__dirname, '../../node_modules')),
```

Comment:
This can get very big, no? It's needed?

Reply:
I may be able to filter it down some to only the cdk libraries (which are needed).